File: 2024/Leetcode/problem41.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            if nums[i] <= 0:
                nums[i] = len(nums) + 1
            
        for val in nums:
            index = abs(val)
            if 1 <= index <= len(nums):
                nums[index - 1] = -1 * abs(nums[index - 1])

        for i in range(len(nums)):
            if nums[i] >= 0:
                return i + 1
        return len(nums) + 1

File: 2024/Leetcode/problem41.ts
function firstMissingPositive(nums: number[]): number {
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] <= 0) {
            nums[i] = nums.length + 1;
        }
    }

    for (let i = 0; i < nums.length; i++) {
        let index = Math.abs(nums[i]);
        if (index <= nums.length) {
            nums[index - 1] = -1 * Math.abs(nums[index - 1]);
        }
    }

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > 0) {
            return i + 1;
        }
    }
    return nums.length + 1;
};